guides/v15-mdc-migration.md

```diff
@@ -0,0 +1,217 @@
+
+```ts
+import {MatLegacyButtonModule} from '@angular/material/legacy-button';
+```
+
+## How to Migrate
+
+Migrating an existing app to use MDC based components is a tool-assisted migration. The Angular
+Material team has built schematics to handle as much of the work as possible. However, some manual
+effort will be needed.
+
+You can reduce the amount of manual effort needed by ensuring that your app follows good practices
+before migrating.
+* Avoid overriding styles on internal Angular Material elements in your CSS as much as possible. If
+  you find yourself frequently overriding styles on internal elements, consider using a component
+  that is designed for more style customization, such as the ones available in the
+  [Angular CDK](/cdk).
+* Use [component harnesses](./using-component-harnesses) to interact with Angular Material
+  components in tests rather than probing internal elements, properties, or methods of the
+  component. Using component harnesses makes your tests easier to understand and more robust to
+  changes in Angular Material
+
+### 1. Update to Angular Material v15
+
+Angular Material now includes a schematic to help migrate pre-existing apps to use the new
+MDC-based components. To get started, upgrade your app to Angular Material 15.
+
+```shell
+ng update @angular/components^15
+```
+
+As part of this update, a schematic will run to automatically move your app to use the "legacy"
+packages containing the old component implementations. This provides a quick path to getting your
+app running on v15 with minimal manual changes.
+
+### 2. Run the MDC Migration Script
+
+After upgrading to v15, you can run the MDC Migration Script to switch from the legacy component
+implementations to the new MDC-based ones.
+
+```shell
+ng generate # TODO(wagnermaciel): Insert command here.
+```
+
+This command will update your app's TypeScript, CSS, and templates to the new implementations,
+updating as much as it can automatically.
+
+#### Running a Partial Migration
+
+Depending on the size and complexity of your app, you may want to migrate a single component (or
+small group of components) at a time, rather than all at once. 
+TODO(wagnermaciel): Add details on this: script params, which components need to move together
+
+You may also want to migrate your app one module at a time instead of all together. It is possible
+to use both the old implementation and new implementation in the same app, as long as they aren't
+used in the same module. TODO(wagnermaciel): Add detail on this: script params.
+
+### 3. Check for TODOs left by the migration script.
+
+In situations where the script is not able to automatically update your code, it will attempt to add
+comments for a human to follow up. These TODO comments follow a common format, so they can be easily
+identified.
+
+```ts
+// TODO(wagnermaciel): Do we have a common format for all TODOs the script adds?
```

Comment:
Yeah, the only todos we add are for css files.

`The following rule targets internal classes of <COMPONENT> that may no longer apply for the MDC version.`